packages/compartment-mapper/src/node-powers.js

```diff
@@ -5,17 +5,42 @@
 /** @typedef {import('./types.js').WritePowers} WritePowers */
 
 /**
- * @param {typeof import('fs')} fs
- * @param {typeof import('crypto')} [crypto]
- * @returns {ReadPowers}
+ * @param {string} location
  */
-export const makeNodeReadPowers = (fs, crypto = undefined) => {
+const fakeFileURLToPath = location => {
+  const url = new URL(location);
+  if (url.protocol !== 'file:') {
+    throw new Error(`Cannot convert URL to file path: ${location}`);
+  }
+  return url.pathname;
+};
+
+/**
+ * @param {string} path
+ */
+const fakePathToFileURL = path => {
+  return new URL(path, 'file://').toString();
+};
+
+/**
+ * @param {Object} args
+ * @param {typeof import('fs')} args.fs
+ * @param {typeof import('url')} [args.url] (required for Windows support)
```

Comment:
Why not make this always mandatory?

Also, should we restrict the `url` module to the bits we're using?

```suggestion
 * @param {Pick<typeof import('url'), 'fileURLToPath' | 'pathToFileURL'>} args.url
```

Reply:
Using `pick` is interesting.

I entertained the idea of making `url` mandatory hereafter. The factoring is a little tricky since `url` must remain optional in the old API for backward compatibility. Perhaps I need a pair of utility methods to launder the type.

Reply:
I was thinking of moving the fake injection stuff in the deprecated methods?

Reply:
Combined with `Pick`, you should have no problem doing:

```js
const fakeUrl = {
  pathToFileURL: () => {},
  fileURLToPath: () => {},
};

export const makeNodeReadPowers = (fs, crypto = undefined) => {
  return makeReadPowers({ fs, url: fakeUrl, crypto });
};

export const makeNodeWritePowers = fs => {
  return makeWritePowers({ fs, url: fakeUrl });
};
```

Reply:
I think I will leave this as-is in order to leave more doors open for future changes. This method attenuates those powers regardless.

Reply:
The change is fine, but I'm not sure I understand the "doors open for future changes". It feels a little weird to have a function which advertises it needs `url` but will in fact be happy without.